### DISCUSSION TITLE
Is it possible to have a Maintenance Branch for the latest release?

### DISCUSSION THREAD

**Discussion by Benuuu** [semantic-release/semantic-release#3078] [Release workflow]
In our workflow, we have a reason to create a maintenance hotfix before we've created the next release from the main branch.

We're running into an issue where semantic-release is saying the following:
```
The release 2.14.1 on branch 2.14.x cannot be published as it is out of range.
Based on the releases published on other branches, only versions within the range >=2.14.0 <2.14.0 can be published from branch 2.14.x.
```

when we have a git history that looks roughly like this:

```
* abcd1234 (origin/master) feat(foo): next-set of feature work
|
| * abcd1234 (origin/2.14.x) fix(bar): fix the stuff that's broken
|/  
* abcd1234 (tag: v2.14.0) chore(release): 2.14.0
```

I think we're in a similar situation to [this recipe step](https://semantic-release.gitbook.io/semantic-release/recipes/release-workflow/maintenance-releases#releasing-a-bug-fix-for-version-2.1.0-users) but we don't want to merge it directly into `master`.

Is this workflow possible with semantic-release? Is this a bug in how the upper part of the range is generated?

**Comment by sheerlox** [comment]
> In our workflow, we have a reason to create a maintenance hotfix before we've created the next release from the main branch.

Could you expand on that? What would be the use case for creating a maintenance branch if you're not behind?

If your goal is to test the hotfix without merging it into the main branch, you should use [a pre-release workflow](https://semantic-release.gitbook.io/semantic-release/recipes/release-workflow/pre-releases), then integrate it back into the main branch as soon as possible.

**Reply by Benuuu** [reply]
We're probably doing semantic-release branching and distribution channels wrong. Basically we create a release on a bi-weekly cadence, then go through some testing as the image gets promoted through various environments. This testing isn't fully automated, so it takes some time. We then deploy some time after that period.

Repeated but in bullet form:
1. Sprint ends, cut release and create image
2. Testing and promotion
3. Deploy
4. Wait for sprint end to repeat

It may be that after 3 and before 1 we discover we need to hotfix.

Beside my use case, why isn't the range consistent with the branch it's on? If I'm creating a release on branch `1.2.x` shouldn't my range be greater than my last `1.2.x` release and less than `1.3`?
